optimizely/client/factory_test.go

```diff
@@ -57,6 +58,26 @@ func TestClientWithSDKKey(t *testing.T) {
 	assert.NotNil(t, optimizelyClient.EventProcessor)
 }
 
+func TestClientWithPollingConfigManager(t *testing.T) {
+	factory := OptimizelyFactory{}
+
+	optimizelyClient, err := factory.Client(WithPollingConfigManager("1212", time.Second, nil))
+	assert.NoError(t, err)
+	assert.NotNil(t, optimizelyClient.ConfigManager)
+	assert.NotNil(t, optimizelyClient.DecisionService)
+	assert.NotNil(t, optimizelyClient.EventProcessor)
+}
+
+func TestClientWithPollingConfigManagerRequester(t *testing.T) {
+	factory := OptimizelyFactory{}
+	requester := utils.HTTPRequester{}
+	optimizelyClient, err := factory.Client(WithPollingConfigManagerRequester(requester, time.Second, nil))
```

Comment:
This has the side effect that it will start the polling and will be polling every second. Since we don't really care about the polling, can we have a mock requester so that we don't send out http requests

Reply:
addresssed